src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+    return false;
+
+  size_t idx = 0;
+  uint16_t crc = 0;
+  static uint16_t rec_buf_len = 0;
+  int command_length = 0;
+#define REC_BUFSIZE 127
+  static uint8_t command[REC_BUFSIZE];
+
+  /** loop as long we have data */
+  while (idx < len)
+  {
+    /** check if we received a header*/
+    if ((uint8_t)string[idx] == 0xA5)
+      rec_buf_len = 0;
+
+    /** check if the buffer-size is reached */
+    if (rec_buf_len >= REC_BUFSIZE)
+      rec_buf_len = 0;
+    assert(rec_buf_len < REC_BUFSIZE);
+
+    /** write command in buffer */
+    command[rec_buf_len++] = (uint8_t)string[idx++];
+    if (rec_buf_len == 2)
+    {
+      /** check if command is PROT-ID */
+      if (!(command[rec_buf_len - 1] == 0x14))
+      {
+        rec_buf_len = 0;
+      }
+    }
+
+    if (rec_buf_len >= 3)
+    {
+      command_length = command[2];
+
+      /** all received */
+      if (rec_buf_len >= (command_length + 5))
+      {
+        /** generate uint16_t out of commands */
+        uint16_t crc_value = BytesToUnsigned(command[command_length + 4], command[command_length + 3]);
+
+        /** generate crc */
+        crc = CRCBitwise(command, command_length + 3);
+
+        /** check crc */
+        if (crc == crc_value || crc_value == 0)
+        {
+          /** extract and save active and passive frequency */
+          SaveAR62xxFrequencies(command, command_length + 5, crc);
+        }
+        rec_buf_len = 0;
+      }
+    }
+  }
+  return true;
+}
+
+/**
+   * @brief save frequencies
+   * 
+   * @param sz_command command which contains the frequencies
+   * @param len length of the command
+   * @param crc CRC
+   */
+void AR62xxDevice::SaveAR62xxFrequencies(uint8_t *sz_command, int len, uint16_t crc)
+{
+  /** check if the command is valid */
+  if (sz_command == NULL || len == 0)
+    return;
+
+  /** command 22 = frequency-settings
+   *  Frequency settings, always for both frequencies (active and passive) */
+  if ((unsigned char)(sz_command[3] & 0x7F) == 22)
+  {
+    active_frequency = AR62xxIdToFrequency(BytesToUnsigned(sz_command[5], sz_command[4]));
+    passive_frequency = AR62xxIdToFrequency(BytesToUnsigned(sz_command[7], sz_command[6]));
+  }
+}
+
+bool AR62xxDevice::PutActiveFrequency(RadioFrequency frequency,
+                                      const TCHAR *name,
+                                      OperationEnvironment &env)
+{
+  /** command */
+  uint8_t szTmp[MAX_CMD_LEN];
+
+  /** generate the command to set the active station */
+  int len = GenerateSetStationCommand(szTmp, ACTIVE_STATION, frequency, (const TCHAR *)name);
+
+  /** send the command */
+  bool isSend = Send((uint8_t *)&szTmp, len, env);
+
+  /** return if command could be send */
+  return isSend;
+}
+
+bool AR62xxDevice::PutStandbyFrequency(RadioFrequency frequency,
+                                       const TCHAR *name,
+                                       OperationEnvironment &env)
+{
+  /** command */
+  uint8_t szTmp[MAX_CMD_LEN] = {};
```

Comment:
Bad variable name.
Why the initializer?

Reply:
removed the initializer

Reply:
renamed variable